modules/codegen/src/main/scala/com/twilio/guardrail/core/Tracker.scala

```diff
@@ -0,0 +1,126 @@
+  implicit def optionaljuListConvincer[A]: Tracker.Convincer[Option[java.util.List[A]], List[A]] = Tracker.Convincer(_.fold(List.empty[A])(_.asScala.toList))
+  implicit def optionalListConvincer[A]: Tracker.Convincer[Option[List[A]], List[A]]             = Tracker.Convincer(_.getOrElse(List.empty[A]))
+  implicit def optionaljuCollectionConvincer[A]: Tracker.Convincer[Option[java.util.Collection[A]], List[A]] =
+    Tracker.Convincer(_.fold(List.empty[A])(_.asScala.toList))
+  implicit def optionalOptionConvincer[A]: Tracker.Convincer[Option[Option[A]], Option[A]] = Tracker.Convincer(_.flatten)
+  implicit def optionaljuMapConvincer[K, V]: Tracker.Convincer[Option[java.util.Map[K, V]], Mappish[List, K, V]] =
+    Tracker.Convincer(_.fold(Mappish(List.empty[(K, V)]))(x => Mappish(x.asScala.toList)))
+  implicit def optionalMapConvincer[K, V]: Tracker.Convincer[Option[Map[K, V]], Mappish[List, K, V]] =
+    Tracker.Convincer(_.fold(Mappish(List.empty[(K, V)]))(x => Mappish(x.toList)))
+}
+
+trait LowestPriorityTrackerInstances {
+  implicit object distributiveTracker extends IndexedDistributive[Tracker] {
+    def indexedDistribute[G[_], A](value: Tracker[G[A]])(implicit G: IndexedFunctor[G]): G[Tracker[A]] =
+      G.map(value.unwrapTracker) {
+        case (i, a) =>
+          new Tracker(a, value.history ++ G.label(i))
+      }
+  }
+
+  implicit def trackerFunctor = new cats.Functor[Tracker] {
+    def map[A, B](fa: Tracker[A])(f: A => B): Tracker[B] = new Tracker(f(fa.unwrapTracker), fa.history)
+  }
+}
+
+trait LowPriorityTrackerSyntax extends LowestPriorityTrackerInstances {
+  implicit class ListSyntax[A](tracker: Tracker[List[A]]) {
+    def toNel: Tracker[Option[NonEmptyList[A]]] = tracker.map(NonEmptyList.fromList _)
+  }
+
+  implicit class MappishSyntax[A, B](tracker: Tracker[Mappish[List, A, B]]) {
+    def toNel: Tracker[Option[Mappish[NonEmptyList, A, B]]] = tracker.map(x => NonEmptyList.fromList(x.value).map(Mappish(_)))
+  }
+}
+
+trait HighPriorityTrackerSyntax extends LowPriorityTrackerSyntax {
+  implicit class StringyEitherSyntax[B](tracker: Tracker[Either[String, B]]) {
+    def raiseErrorIfLeft: Target[Tracker[B]] = tracker.fold(err => Target.raiseError(s"${err.get} (${err.showHistory})"), Target.pure _)
+  }
+
+  implicit class EitherSyntax[A, B](tracker: Tracker[Either[A, B]]) {
+    def fold[C](a: Tracker[A] => C, b: Tracker[B] => C): C =
+      tracker.unwrapTracker.fold(x => a(Tracker.cloneHistory(tracker, x)), x => b(Tracker.cloneHistory(tracker, x)))
+  }
+
+  implicit class OptionSyntax[A](tracker: Tracker[Option[A]]) {
+    def orHistory: Either[Vector[String], Tracker[A]]      = tracker.indexedCosequence.toRight(tracker.history)
+    def raiseErrorIfEmpty(err: String): Target[Tracker[A]] = Target.fromOption(tracker.indexedCosequence, s"${err} (${tracker.showHistory})")
+    class FlatDownFieldPartiallyApplied[C](val dummy: Boolean = true) {
```

Comment:
`extends AnyVal`?  (Otherwise, why bother with `dummy`?)

(& other examples of the same in this file)

Reply:
This follows the pattern used in cats to partially apply type arguments. `dummy` is presumably required to avoid syntactic ambiguity, but the big thing is by fixing `C` in the call site, it permits:

`t.downField[Int]("foo", _.getFoo)` instead of `t.downField[Option[Int], Int]("foo", _.getFoo)`

Small improvement in ergonomics for the user, as it frees the unspecified type parameter to be resolved by the compiler.

Reply:
Yeah, I've seen that pattern cats, but they do `extends AnyVal` on those.  I assumed the need for `dummy` was just because you have to have one-and-only-one `val` constructor param when extending `AnyVal`.